src/depth-to-rgb-calibration.cpp

```diff
@@ -120,8 +138,22 @@ rs2_calibration_status depth_to_rgb_calibration::optimize(
             AC_LOG( DEBUG, DISABLE_RS2_CALIBRATION_CHECKS << " is on; continuing" );
         }
 
+        if (!should_continue())
+        {
+            call_back(RS2_CALIBRATION_FAILED);
+            return RS2_CALIBRATION_FAILED;
+        }
+
         AC_LOG( DEBUG, "... optimizing" );
-        _algo.optimize();
+
+        std::function<void(impl::data_collect const &)> opt_cb ([&](impl::data_collect const &data) 
```

Comment:
May as well just make it 'auto' so don't have to repeat the definition of the function

Reply:
Done.